### PR TITLE
fix: settings allowed values check

### PR DESCRIPTION
**Pull request by hpohekar**
closes https://github.com/ansys/pyfluent/issues/2778

1. Updated attribute check in settings exception block.

### CODE REVIEW

File: src/ansys/fluent/core/solver/flobject.py [hunk 1]
```diff
@@ -1067,9 +1067,10 @@ def __setattr__(self, name: str, value):
...
         except Exception as ex:
-            allowed = attr.allowed_values()
-            if allowed and value not in allowed:
-                raise allowed_values_error(name, value, allowed) from ex
+            if hasattr(attr, "allowed_values"):
+                allowed = attr.allowed_values()
+                if allowed and value not in allowed:
+                    raise allowed_values_error(name, value, allowed) from ex
             else:
                 raise ex
```
Comment (h-krishnan) [inline src/ansys/fluent/core/solver/flobject.py:1075]: Wouldn't a plain `raise` be sufficient here?

**Review by h-krishnan — APPROVED** [review]

File: src/ansys/fluent/core/solver/flobject.py (hunk 1 above)
Comment (hpohekar) [reply src/ansys/fluent/core/solver/flobject.py:1075]: @h-krishnan Yes, but we think that it looks safe.

Comment (hpohekar) [reply src/ansys/fluent/core/solver/flobject.py]: Done. Thank you.

**Review by seanpearsonuk — APPROVED** [review]

**Comment by seanpearsonuk** [comment]
@hpohekar please add a test. If you get stuck, please check with @h-krishnan. Please ensure your test fails with the original code. The test is important as we would be patching the product with this.

**Review by seanpearsonuk — APPROVED** [review]

**Review by prmukherj — APPROVED** [review]

**Review by mkundu1 — APPROVED** [review]

**Pull request merged**